questions/javascript-questions.md

```diff
@@ -64,3 +64,4 @@ duplicate([1,2,3,4,5]); // [1,2,3,4,5,1,2,3,4,5]
 * What are the benefits of using `spread syntax` and how is it different from `rest syntax`?
 * How can you share code between files?
 * Why you might want to create static class members?
+* After how much time will the `callback` be executed if invoked as: `setTimeout(callback, 4000)`?
```

Comment:
looks like `after what time` should be

Reply:
and order of words

Reply:
Maybe `When will the callback be executed ...` ?

Reply:
It should be executed at time equivalent to 4000 milli seconds or more not at 3999 milliseconds.

Reply:
I often ask a question about setTimout, in the context of a question about arrow functions. I first ask the interviewee to explain what setTimeout is and then I ask the following, which opens up a larger, potentially open-ended discussion. Trivia questions with a "right" answer are the kinds of things that people can cram for (by looking at a site like this.) Questions that serve as jumping off points are much less easy to prepare for, in advance. 

```suggestion
* When would you expect the `callback` function in the following setTimeout to be executed? 
`setTimeout(callback, 0)`?
```